capnp/helpers/capabilityHelper.cpp

```diff
@@ -0,0 +1,163 @@
+#include "capnp/helpers/capabilityHelper.h"
+#include "capnp/lib/capnp_api.h"
+
+::kj::Promise<PyObject *> convert_to_pypromise(capnp::RemotePromise<capnp::DynamicStruct> & promise) {
+    return promise.then([](capnp::Response<capnp::DynamicStruct>&& response) { return wrap_dynamic_struct_reader(response); } );
+}
+
+void reraise_kj_exception() {
+  GILAcquire gil;
+  try {
+    if (PyErr_Occurred())
+      ; // let the latest Python exn pass through and ignore the current one
+    else
+      throw;
+  }
+  catch (kj::Exception& exn) {
+    auto obj = wrap_kj_exception_for_reraise(exn);
+    PyErr_SetObject((PyObject*)obj->ob_type, obj);
+    Py_DECREF(obj);
+  }
+  catch (const std::exception& exn) {
+    PyErr_SetString(PyExc_RuntimeError, exn.what());
+  }
+  catch (...)
+  {
+    PyErr_SetString(PyExc_RuntimeError, "Unknown exception");
+  }
+}
+
+void check_py_error() {
+    GILAcquire gil;
+    PyObject * err = PyErr_Occurred();
+    if(err) {
+        PyObject * ptype, *pvalue, *ptraceback;
+        PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+        if(ptype == NULL || pvalue == NULL || ptraceback == NULL)
+          throw kj::Exception(kj::Exception::Type::FAILED, kj::heapString("capabilityHelper.h"), 44, kj::heapString("Unknown error occurred"));
+
+        PyObject * info = get_exception_info(ptype, pvalue, ptraceback);
+
+        PyObject * py_filename = PyTuple_GetItem(info, 0);
+        kj::String filename(kj::heapString(PyBytes_AsString(py_filename)));
+
+        PyObject * py_line = PyTuple_GetItem(info, 1);
+#if PY_MAJOR_VERSION >= 3
```

Comment:
We can just remove the Python 2 code here as v1.0.0 drops Python 2.7 support.

Reply:
Done